File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int number_o_letters(string txt);
int number_o_sentences(string txt);
int number_o_words(string txt);

int main(void)
{
    // get input from user
    string text = get_string("Text:");
    // score the whole text
    int l = number_o_letters(text);
    int s = number_o_sentences(text);
    int w = number_o_words(text);
    float avg_l = (float) l / w * 100;
    float avg_s = (float) s / w * 100;
    int index = (int) (0.0588 * avg_l - 0.296 * avg_s - 15.8 + 0.5);

    // Print the readability level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}

int number_o_letters(string txt)
{
    int count_l = 0;
    for (int i = 0, n = strlen(txt); i < n; i++)
    {
        char alpha = txt[i];
        if (isalpha(alpha))
        {
            count_l += 1;
        }
    }
    return count_l;
}
int number_o_sentences(string txt)
{
    int count_s = 0;
    for (int i = 0, n = strlen(txt); i < n; i++)
    {
        char alpha = txt[i];
        if (alpha == '!' || alpha == '.' || alpha == '?')
        {
            count_s += 1;
        }
        else
        {
            count_s += 0;
        }
    }
    return count_s;
}
int number_o_words(string txt)
{
    int count_w = 0;
    for (int i = 0, n = strlen(txt); i < n; i++)
    {
        char alpha = txt[i];
        if (alpha == ' ')
        {
            count_w += 1;
        }
    }
    count_w += 1;
    return count_w;
}

File: sentimental-readability/readability.py
from cs50 import get_string

# Get user input
txt = get_string("Text: ")

def count_letters(text):
    return sum(1 for char in text if char.isalpha())

def count_words(text):
    return len(text.split())

def count_sentences(text):
    return sum(1 for char in text if char in ['.', '?', '!'])

count_l = count_letters(txt)
count_w = count_words(txt)
count_s = count_sentences(txt)

# Handle potential division by zero
if count_w == 0:
    index = 0
else:
    L = (count_l / count_w) * 100
    S = (count_s / count_w) * 100
    index = 0.0588 * L - 0.296 * S - 15.8

# Format the result to one decimal place
formatted_index = round(index, 1)

if formatted_index < 1:
    print("Before Grade 1")
elif formatted_index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {formatted_index}")



